src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,344 @@
+        """
+        Acquires image data from the spectrometer.
+        """
+        self.file_setup()
+        self.start_acquisition_and_wait()
+        return self.process_acquired_data()
+
+    def close(self) -> None:
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self._automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+class SpectrometerConfig():
+
+    MIN_WAVELENGTH_DIFFERENCE = 117
+
+    def __init__(self, experiment_name=None) -> None:
+        self._experiment_name = experiment_name
+        self.light = LightfieldApplicationManager(True)
+
+    def finalize(self) -> None:
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self.light.close()
+    
+    def get_wavelengths(self) -> np.ndarray:
+        """
+        Returns the wavelength calibration for a single frame.
+
+        According to the person who made the Lightfield sofware there is not current way to use the exact Lightfield settings
+        for step and glue. Will have to interpolate for now.
+        """
+        wavelength_array = np.fromiter(self.light.experiment.SystemColumnCalibration, np.float64)
+        return wavelength_array
+    
+    @property
+    def center_wavelength(self) -> float:
+        """
+        Returns the center wavelength in nanometers.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingCenterWavelength)
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers: float) -> None:
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        # The line below addresses bug where if step and glue is enabled it wont allow you to set the center wavelength.
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+        self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, nanometers)
+
+    @property
+    def experiment_name(self) -> Union[str, None]:
+        """
+        Returns the experiment name.
+        """
+        return self._experiment_name
+
+    @experiment_name.setter
+    def experiment_name(self, a_name: str) -> None:
+        """
+        User can set the experiment that they want to load.
+        """
+        if a_name != self._experiment_name:
+            # Added this to prevent errors with incorrect file name
+            if a_name in self.light.experiment.GetSavedExperiments(): 
+                self._experiment_name = a_name
+                self.light.load_experiment(self._experiment_name)
+            else:
+                logger.error(f"An experiment with that file name does not exist.")
+
+    @property
+    def grating(self) -> str:
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, grating_string: str) -> None:
```

Comment:
It seems pretty standard throughout python projects to use "value" as the input variable for a setter

```
def grating(self, value: str) -> None:
   ...
```